tests/Doctrine/Tests/DoctrineTestCase.php

```diff
@@ -4,9 +4,59 @@
 
 namespace Doctrine\Tests;
 
+use BadMethodCallException;
+use PHPUnit\Framework\TestCase;
+
+use function get_called_class;
+use function sprintf;
+
 /**
  * Base testcase class for all Doctrine testcases.
+ *
+ * This includes polyfillable logic for PHPUnit compatibility.
  */
-abstract class DoctrineTestCase extends PolyfillableTestCase
+abstract class DoctrineTestCase extends TestCase
 {
+    /**
+     * @param array<mixed> $arguments
+     *
+     * @return mixed
+     */
+    public static function __callStatic(string $method, array $arguments)
+    {
+        if ($method === 'assertMatchesRegularExpression') {
+            return self::assertRegExp(...$arguments);
+        } elseif ($method === 'assertDoesNotMatchRegularExpression') {
```

Comment:
On the specific thing of how you support older phpunit versions, my thoughts are:

* Having to duplicate these stubs between `__callStatic` and `__call` is not ideal
* It's not at all clear which phpunit versions these are needed for, so although it's *possible* to remove them by editing a single file it's fairly likely one or both *won't* actually be removed when a given phpunit version is dropped. That could be helped by at least commenting each one to document the phpunit version they're for.
* It boxes you into a corner if a future phpunit release includes something that can't be fixed just by proxying a method with the same argument sequence.

More importantly, having the polyfill present at runtime in newer phpunit versions increases the risk for confusion & unexpected behaviour in future. When you eventually update to e.g. phpunit 10, it will be much easier to find all the things that need to be updated by iterating till you get a green run on the new version, then adding polyfills on the old versions if / as required. Otherwise there's a high chance some method calls in tests are missed, and don't break till you drop the old version some years later when people will have forgotten what they need to be changed to. 

It also allows contributors to add tests with outdated syntax, which just adds to the work of dropping the old version / polyfill eventually.

Or - worst case - the polyfill doesn't do what you expect (at first, or over time), so you get false positives on tests that should be failing, and would be if they were calling the (tested) phpunit methods directly.